### ISSUE TITLE
some dependencies can be devDependencies

### ISSUE THREAD

**Issue by david-fong** [lcheng1/decabyt.es#22]
Hello there and nice repo!

I noticed that your package.json's dependencies include webpack (plugins, loaders, etc.) and eslint packages. I'm not familiar with your build and deployment process, but I know that those tend to be devDependencies. Is this intentional?

**Comment by lcfyi** [comment]
Hey David, it is intentional. It's the unfortunate result of my CD pipeline.

I had originally wanted this project to be an npm package that handled the terminal interface for you (ie. you can inject your own config, and this package would handle all the I/O), but I reworked it when I realized I didn't actually want to publish a library. The rework resulted in the current plugin architecture that's dynamically initialized during the build process.

Currently, I use Netlify for the build and deployment but their CI/CD doesn't include the ability to define stages. Thus, I needed to chain commands together to simulate a test/build pipeline (ie. the testing and build stages shared the same dependencies). I had also originally wanted a pre-prod environment that I could deploy to, but when I tried setting `NODE_ENV` flags to differentiate the environments, the `devDependencies` wouldn't install on prod causing the chained build process to fail.

At that point I just shifted all the previous `devDependencies` into regular dependencies since it didn't make a huge difference for my deployment or development. I also discovered that Netlify's local toolchain worked pretty well so I abandoned bothering to set up stages. ¯\\\_(ツ)\_/¯

tl;dr: yes, they can be `devDependencies` as long as `NODE_ENV` is not set to `production`.

**Comment by david-fong** [comment]
Hey Leslie! I see! Thanks for the comprehensive explanation. I really appreciate you writing that all out :) Feel free to close this issue.
